### PR TITLE
Fix issue with Python version breaking homebrew release

### PR DESCRIPTION
**Pull request by spencerdrak**
PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -302,15 +302,23 @@ jobs:
     name: Update on Homebrew-Core
     needs: [sleep-before-homebrew] # Needs to run after pypi released so brew can update pypi dependency hashes
     runs-on: macos-10.15
-    strategy:
-      matrix:
-        python: ["3.9.12"]
     steps:
       - name: Setup Python
         uses: actions/setup-python@v1
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
        uses: actions/setup-python@v3
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -302,15 +302,23 @@ jobs:
...
         with:
-          # Needs to be < 3.10 see https://github.com/returntocorp/semgrep/issues/4213
-          python-version: ${{ matrix.python }}
+          # Pin to python 3.9 for pipgrip issues: see https://github.com/returntocorp/semgrep/issues/4213
+          python-version: '3.9.*'
```
Comment (underyx) [inline .github/workflows/release.yml]:
```suggestion
          python-version: '3.9.x'
```

I'm sure that worked as well, but this format matches the documentation at https://github.com/actions/setup-python

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -302,15 +302,23 @@ jobs:
...
+        id: python-version
+        run: |
+          # older versions of python pass the version output on stderr, redirect this to stdout
+          # Input: Python 3.9.12
+          # Output: 3.9.12
+          # sed -n: only print match
+          # [Pp]ython : consume initial output up to first digit
+          # [0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*: match a pattern of three groups of one or more digits and dots, using basic RE
+          PYTHON_VERSION=$(python --version 2>&1 | sed -n 's|[Pp]ython \([0-9][0-9]*.[0-9][0-9]*.[0-9][0-9]*\)*$|\1|p')
+          echo ::set-output name=python_version::"$PYTHON_VERSION"
```
Comment (underyx) [inline .github/workflows/release.yml]: Seems like we can actually get rid of all this code cause setup-python outputs the version 🎉 https://github.com/actions/setup-python/issues/235

So something like `${{ steps.setup-python.outputs.python-version }}` should work already if we set a step ID for it.

File: .github/workflows/release.yml (hunk 1 above)
Comment (spencerdrak) [reply .github/workflows/release.yml]: change applied

File: .github/workflows/release.yml (hunk 2 above)
Comment (spencerdrak) [reply .github/workflows/release.yml]: change applied

File: .github/workflows/release.yml (hunk 3 above)
Comment (spencerdrak) [reply .github/workflows/release.yml]: change applied

**Review by underyx — APPROVED** [review]

**Pull request merged**